server/openai/generateItenerary.js

```diff
@@ -0,0 +1,56 @@
+const openaiClient = require('./openaiClient')
+
+async function generateItinerary(constraints) {
+  try {
+    const conversation = [
+      {
+        role: 'system',
+        content: `You are an AI that generates travel itineraries. Respond with ONLY JSON that contains the plan for each stage. 
+        ONLY respond with the following format, do not include any descriptions or codeblocks, I should be able to parse the output to a JavaScript Object
+        The response needs to be formatted exactly like the following structure. Add some Food and snack stages as well with an optional: true marker.:
+        '''
+        {
+          plan:[
+            {
+              day: Number
+              bullet_point_summary: String (comma seperated)
+              stages:[
+                {
+                  stage: Number,
+                  place: String,
+                  description: String,
+                  address: String,
+                  emoji: String (best emoji representation of stage)
+                }
+              ]
+            }
+          ]
+        }
+        '''
+        
+        If there are any confusing values respond with:
+        '''
+        {
+          error: String (reason)
+        }
+        '''
+        `,
+      },
+      {
+        role: 'user',
+        content: `Generate a travel itinerary based on these constraints: ${constraints}`,
+      },
+    ]
+
+    const response = await openaiClient(conversation)
+    if (response) {
+      return response
+    }
+    return 'Unable to generate a travel itinerary. Please try again later.'
+  } catch (error) {
```

Comment:
a little confused about the error response described in lines 33-35: do we only get that value from `response.error`? If so, can we handle that here too and throw it as an error? 

The error you're catching here looks separate from that, but just wanted to confirm (am on mobile so can't test)

Reply:
this can probably be written better but there are errors at play here. One is an error thrown by the openaiClient and the second is the error message generated by the API. 

the error thrown by openai client is caught and processed. 

We would need seperate logic to handle the JSON error response. It would indicate that one of the values was incorrectly passed for example: I pass Romeo for the Destination instead of Rome

Reply:
I see, thanks for clarifying. opened a new comment above for discussion